pallets/settlement/src/lib.rs

```diff
@@ -723,7 +607,7 @@ decl_module! {
         /// # Arguments
         /// * `instruction_id` - Instruction id for that affirmation get withdrawn.
         /// * `portfolios` - Portfolios that the sender controls and wants to withdraw affirmation.
-        #[weight = 25_000_000_000]
+        #[weight = <T as Trait>::WeightInfo::withdraw_affirmation(portfolios.len() as u32)]
         pub fn withdraw_affirmation(origin, instruction_id: u64, portfolios: Vec<PortfolioId>) -> DispatchResult {
```

Comment:
How are you limiting legs here? One portfolio can be involved in multiple legs and the weight should be a function of the number of legs, not the number of portfolios.

Reply:
Same as above

Reply:
(Closing, let's continue the discussion at a single place)